lib/submit-analytics.js

```diff
@@ -33,50 +104,184 @@ function sha256(data) {
-                        'properties': {
-                            'token': 'aab85907a13e1ff44a95be539d9942a9',
-                            'distinct_id': identifier,
-                            'Anonymized Machine Identifier': identifier,
-                            'Anonymized Application ID': sha256(__dirname),
-                            'Binding': context || 'node.js',
-                            'Version': require('../package.json').version,
-                            'Language': 'javascript',
-                            'OS Type': os.platform(),
-                            'OS Version': os.release(),
-                            'Node.js versions': process.versions
-                        }
-                    };
-
-                    request(`https://api.mixpanel.com/track/?data=${Buffer.from(JSON.stringify(payload), 'utf8').toString('base64')}&ip=1`,
-                    () => { /* Analytics failed. Do nothing. */ });
+                if (versionTags[0] < 10) {
+                    // on v6.x, we submit both MixPanel and S3 analytics
+                    // MixPanel analytics make no distinction between production and staging modes
+                    dispatchAnalytics(new mixpanelDetails(), payload);
+                }
+
+                dispatchAnalytics(requestFactory, payload)
+                .catch((e) => {
+                    reject(e);
+                    return;
+                })
+                .then(() => {
+                    resolve('');
+                    return;
                 });
+            });
         }, 100);
-    }
 
-    if (require.main === module) {
-        module.exports('Install');
-    }
+    }); // end Promise
+}
+
+/**
+ * Shorthand for submitting analytics to production environment
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns Promise
+ */
+async function submitProductionAnalytics(eventName) {
+    const context = require('../package.json')
+    return submitAnalytics(webhookProdDetails, context, eventName);
+}
+
+/**
+ * Shorthand for submitting analytics to staging environment
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns Promise
+ */
+async function submitStageAnalytics(eventName) {
+    const context = require('../package.json');  // Realm context
+    return submitAnalytics(webhookStageDetails, context, eventName);
+}
+
+module.exports.fetchPlatformData = fetchPlatformData;
+module.exports.webhookProdDetails = webhookProdDetails;
+module.exports.webhookStageDetails = webhookStageDetails;
+module.exports.submitAnalytics = submitAnalytics;
+module.exports.submitProductionAnalytics = submitProductionAnalytics;
+module.exports.submitStageAnalytics = submitStageAnalytics;
+module.exports.isAnalyticsDisabled = isAnalyticsDisabled;
+
+if (require.main === module) {
+    module.submitProductionAnalytics('Install');
```

Comment:
Little need to call this off the `module` global, since it's already a local variable in scope here.

```suggestion
    submitProductionAnalytics('Install');
```